tests/components/sensor/test_init.py

```diff
@@ -546,6 +547,31 @@ async def test_custom_unit(
             1000,
             SensorDeviceClass.ENERGY,
         ),
+        # Power factor
+        (
+            None,
+            PERCENTAGE,
+            PERCENTAGE,
+            1.0,
+            100,
+            SensorDeviceClass.POWER_FACTOR,
+        ),
+        (
+            PERCENTAGE,
+            None,
+            None,
+            100,
+            1,
+            SensorDeviceClass.POWER_FACTOR,
+        ),
+        (
+            "Cos φ",
```

Comment:
Side note: Cos Phi is not the unit of measurement for power factor. It's another way of naming the power factor in certain systems.

Reply:
That's right, the reason for including it here is that `Cos φ` is incorrectly used as the unit for power factor by some integrations (probably as a workaround for the previous limitation where frontend would not display a graph for sensors with no unit).